### PR TITLE
Updated the site api endpoint

### PR DESCRIPTION
**Pull request by enejb**
Update to be the same as the recent changes on .com to the setting api endpoint. 

To test
Go to https://developer.wordpress.com/docs/api/console/

and request the endpoint 
/sites/$siteID 

Notice that it returns the same data as /me/sites for a particular site. 

Change constants to something else. 
Retest. 

cc: @roccotripaldi and @lezama 

I think this doesn't quite work as expected yet.

### CODE REVIEW

**Comment by enejb** [comment]
@lezama and @roccotripaldi can you test this to make sure we get what we expect here? And also make sure that the code makes sense? 

To test on https://developer.wordpress.com/docs/api/console/
I tested the `/sites/123456789` api endpoint. 

having this defined in wp-config.php
define( 'FS_METHOD', 'ssh2' );
define( 'DISALLOW_FILE_EDIT', true );
define( 'DISALLOW_FILE_MODS',true );
define( 'WP_AUTO_UPDATE_CORE', false );
define( 'AUTOMATIC_UPDATER_DISABLED', true );

gives me 
screen shot 2015-11-12 at 4 31 06 pm

File: class.jetpack-sync.php [hunk 1]
```diff
@@ -1030,6 +1053,19 @@ function mock_option( $option , $callback ) {
 	function register_constant( $constant ) {
 		$this->register( 'constant', $constant );
 	}
+
+	function get_default_constant() {
+		$filter = current_filter();
+		// We don't know what the constant is so we get it from the current filter.
+		if ( substr( $filter, 0, 28 ) === 'pre_option_jetpack_constant_' ) {
```
Comment (dereksmart) [inline class.jetpack-sync.php]: :yoda: style

File: class.jetpack-sync.php [hunk 2]
```diff
@@ -1030,6 +1053,19 @@ function mock_option( $option , $callback ) {
...
+			$constant = substr( $filter, 28 );
+			if ( defined( $constant ) ) {
+				// We if constant is set to false we will not shortcut the get_option function and will return the default value.
```
Comment (dereksmart) [inline class.jetpack-sync.php]: "We if constant..." sounds like typo

File: class.jetpack-sync.php [hunk 3]
```diff
@@ -825,14 +825,38 @@ function sync_all_constants() {
...
+			add_filter( 'pre_option_jetpack_constant_'. $constant, array( $this, 'get_default_constant' ) );
+		}
+	}
 
+	function sync_all_constants() {
 		// add the constant to sync.
-		foreach( $constants as $contant ) {
-			$this->register_constant( $contant );
+		foreach( $this->get_all_constants() as $constant ) {
+			$this->register_constant( $contsant );
```
Comment (roccotripaldi) [inline class.jetpack-sync.php]: typo! :scream:

**Comment by roccotripaldi** [comment]
Other than the typo in the variable, this is looking good.

I've tested on a few different sites, with different combinations of defined constants. It is working as expected.

File: class.jetpack-sync.php (hunk 3 above)
Comment (enejb) [inline class.jetpack-sync.php]: Good catch!

**Comment by enejb** [comment]
Fixed the typo and hopefully clarified the comment.

**Comment by enejb** [comment]
@dereksmart just updated to use Yoda style in the if statement

**Comment by dereksmart** [comment]
LGTM, thanks @enejb

**Pull request merged**